src/oscar/apps/catalogue/views.py

```diff
@@ -41,6 +41,10 @@ def get(self, request, **kwargs):
         if redirect is not None:
             return redirect
 
+        # Do allow staff members so they can test layout etc.
+        if not product.is_public and not request.user.is_staff:
```

Comment:
To make this logic easier to override I think we should use a separate method, e.g., `self.is_viewable()`.

Reply:
feedback applied